torchtitan/experiments/kernels/moe/indices.py

```diff
@@ -170,19 +186,27 @@ def generate_permute_indices(
         To:   | E0 | E1 | E2 | E3 | E0 | E1 | E2 | E3 |
               |  4 |  2 |  1 |  3 |  1 |  2 |  3 |  4 |
     """
+
     # prefix sum to get start index of each expert (parallel scan kernel in future?)
     start_index_values = (
         torch.cumsum(tokens_per_expert_group, 0) - tokens_per_expert_group
     )
 
-    # chunk sizes for each expert
-    chunk_size_per_expert = tokens_per_expert_group.view(num_ranks, -1).sum(0)
+    # total tokens for each expert (sum over ranks)
+    total_tokens_per_expert = tokens_per_expert_group.view(num_ranks, -1).sum(0)
 
-    # align the chunk sizes (cdiv)
-    m_sizes = ((chunk_size_per_expert + alignment - 1) // alignment * alignment).to(
-        torch.int32
```

Comment:
Alternatively, you can just remove the "-1" here, which will also resolve the issue -- 
0 will be up-aligned to 8;
1-7 will be up-aligned to 8, same as before;
8 will be up-aligned to 16, a little bit overhead, but would be okay if the token len is a large enough number.

Reply:
I don't think we need to pad 8 to 16 etc, better to just pad 0.

Reply:
agree though it's a viable option I hadn't thought of generically, let's close this option and focusing on the cleaner no intermediate tensor option above (is_zeros.int() or torch.where).